biojava-core/src/main/java/org/biojava/nbio/core/search/io/Hsp.java

```diff
@@ -125,19 +129,24 @@ public boolean equals(Object obj) {
         return returnAln;

```

Comment:
The above assignments to alignedQuery/Hit need `<>` added so the types get inferred. This works if getSequence returns a `S` as it should.